src/BoundsInference.cpp

```diff
@@ -328,10 +328,9 @@ class BoundsInference : public IRMutator {
                 }
             }
 
-            const vector<Specialization> &specializations = def.specializations();
-            for (size_t i = specializations.size(); i > 0; i--) {
-                Expr s_cond = specializations[i - 1].condition;
-                const Definition &s_def = specializations[i - 1].definition;
+            for (const auto &s : def.specializations()) {
```

Comment:
Are you sure it's ok to reverse the order of results here? The ordering leaks out into all the surrounding code.

Reply:
Oops. Fixed.